### PR TITLE
Add modals for properties with valueTemplateRefs 

### PR DESCRIPTION
**Pull request by ndushay**
To review:
- take a careful look at existing components I *changed*:
    - ResourceTemplate  (replaced FormWrapper with ResourceTemplateForm, improved tests)
    - sinopiaServerSpoof  (added barcode rt to quiet console errors during testing)
    - MonographInstance (changed barcode rtId prefix)
- look at the new classes carefully
     - ResourceTemplateForm renders
     - ModalToggle renders
     - ResourceTemplateModal 
- think about whether I am testing the new classes sufficiently

If there's objection to the removal of unused components, I can take those commits out of this PR.

Known problems/todos:
- integration tests (#223)
- need to accommodate these this:  `repeatable` (#217), `mandatory` (#218), `editable` and `defaultValue` (#219 for both)
- use propTypes.shape for more complex objects in arrays (part of #211)
- ~~ensure valueTemplateRefs are always resourceTemplate ids (and not vocabularies, etc) (#220)~~
- holding the data from a ResourceTemplateModal  (once we decide how we will be modelling that:  redux or ...) (#221)
- styling of buttons, etc:  (#222)

Editor spins up like this:

image

and if you click a button (e.g. "Title"), you get something like this:

image

and if you click the "Title note" button, you get this:

image

Accomplished by replacing FormWrapper component with ResourceTemplateForm component, which is reused in the Modals.

### CODE REVIEW

File: src/components/editor/ResourceTemplateModal.jsx [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+    this.props.toggleVisibility()
+  }
+
+  handleSave = () => {
+    this.saveDataFromModal()
+    this.handleClose()
+  }
+
+  saveDataFromModal = () => {
+    console.log('TODO: implement holding data from ResourceTemplateModal')
```
Comment (SaravShah) [inline src/components/editor/ResourceTemplateModal.jsx]: Should we create a ticket # for this?

File: src/components/editor/ModalToggle.jsx [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+    this.toggleModal = this.toggleModal.bind(this)
+    this.state = {
+      visible: false
+    }
+  }
+
+  toggleModal = () => {
+    const prevState = this.state
+    prevState.visible = !prevState.visible
+    this.setState(prevState => (prevState))
```
Comment (SaravShah) [inline src/components/editor/ModalToggle.jsx]: i think you could also do this if you want: 

```
this.setState({
    visible: !this.state.visible
})
```

File: __tests__/components/editor/ResourceTemplateForm.test.js [hunk 3]
```diff
@@ -0,0 +1,94 @@
...
+    {
+      "propertyLabel": "Literally",
+      "type": "literal"
+    },
+    {
+      "propertyLabel": "Look up, look down",
+      "type": "lookup"
+    },
+    {
+      "propertyLabel": "Bullseye!",
```
Comment (SaravShah) [inline __tests__/components/editor/ResourceTemplateForm.test.js]: loving the `propertyLabels`

File: src/components/editor/ModalToggle.jsx (hunk 2 above)
Comment (ndushay) [reply src/components/editor/ModalToggle.jsx]: I tried some stuff;   dunno if I tried this.

File: static/spoofedFilesFromServer/fromSinopiaServer/resourceTemplates/MonographInstance.json [hunk 4]
```diff
@@ -175,7 +175,7 @@
         },
```
Comment (jermnelson) [inline static/spoofedFilesFromServer/fromSinopiaServer/resourceTemplates/MonographInstance.json]: This is not specific to this code, but I realized another source we should use for determining such variables as `repeatable` and `mandatory` is the Library of Congress Bibframe documentation at https://www.loc.gov/bibframe/. For example, in this BIBFRAME file the 'instanceOf` propertyTemplate has false for repeatable, but I as understand BIBFRAME 2.0 ontology, at this level of modeling, Instance should have only one Work, while Works have multiple Instances.

File: src/components/editor/ResourceTemplateModal.jsx (hunk 1 above)
Comment (jermnelson) [reply src/components/editor/ResourceTemplateModal.jsx]: Yeah, I think this is a candidate for using Redux.

File: __tests__/components/editor/ResourceTemplateForm.test.js [hunk 5]
```diff
@@ -0,0 +1,94 @@
...
+
+  describe('buttons for resource template modals (b/c we have a property of type "resource" with correct valueTemplateRefs)', () => {
+    it('renders ButtonToolbar containing ModalToggle(s)', () => {
+      expect(wrapper
+        .find('div.ResourceTemplateForm ButtonToolbar ModalToggle').length)
+        .toEqual(2)
+    })
+    it('passes appropriate props to ModalToggle(s)', () => {
+      wrapper.find('ModalToggle').forEach((node) => {
+        expect(node.prop('buttonLabel')).toEqual('Note') // TODO: ensure we get from spoofed resourceTemplate
```
Comment (jgreben) [inline __tests__/components/editor/ResourceTemplateForm.test.js]: Maybe say a bit more about this TODO or issue#?

**Comment by ndushay** [comment]
Josh, et al:

yes, sorry about the length.  At least I removed as many files as I added - heh.

I made a separate ticket for the integration tests(#223);  hopefully this is one of the largest PRs we'll have to do for a long time.  (cough cough).   I think whenever possible we should include integration tests with the PR ... if we think the UI is settled down enough.  I already made a ticket for the "backlog" of missing integration tests (#208): and Sarav made a ticket for the InputLiteral (#212), so ... ahem.

 :-)

File: __tests__/components/editor/ResourceTemplateForm.test.js (hunk 5 above)
Comment (ndushay) [reply __tests__/components/editor/ResourceTemplateForm.test.js]: haha - I have no idea what I was thinking when I wrote that TODO comment.  I will remove it.

**Pull request merged**